core/domain/suggestion_registry.py

```diff
@@ -421,24 +424,26 @@ def __init__(
             exp_domain.ExplorationChange(change)
         )
         self.score_category = score_category
-        # In BaseSuggestion, language_code is defined with only string type
-        # but here language_code is of Optional[str] type because language_code
-        # can accept None values as well. So, due to this conflict in types
-        # MyPy throws an `Incompatible types in assignment` error. Thus to
-        # avoid the error, we used ignore here.
+        # Here we use MyPy ignore because in BaseSuggestion, language_code
+        # is defined with only string type but here language_code is of
+        # Optional[str] type because language_code can accept None values as
+        # well. So, due to this conflict in types MyPy throws an `Incompatible
+        # types in assignment` error. Thus to avoid the error, we used ignore.
         self.language_code = language_code  # type: ignore[assignment]
-        # In BaseSuggestion, last_updated is defined with only datetime type
-        # but here last_updated is of Optional[datetime] type because
-        # last_updated can accept None values as well. So, due to this conflict
-        # in types MyPy throws an `Incompatible types in assignment` error.
-        # Thus to avoid the error, we used ignore here.
+        # Here we use MyPy ignore because in BaseSuggestion, last_updated
```

Comment:
Same thing here ... but what does it mean for last_updated to be None? I thought last_updated is always set (it's first equal to the created_on date and then updated to later dates thereafter) so could you explain which conditions cause last_updated to be equal to None?

Reply:
In all suggestion classes, last_updated is defined as a keyword argument with the default value as None ( this is also a reason why we have to use Optional in typing ). Also, while creating suggestions we are not providing last_updated argument,
https://github.com/oppia/oppia/blob/68bb129c7fe773b989670a59162534b8893789bb/core/domain/suggestion_services.py#L224
So, None is used for language_code.

Reply:
comment added!